LEGO1/lego/legoomni/src/race/jetskirace.cpp

```diff
@@ -1,28 +1,153 @@
 #include "jetskirace.h"
 
+#include "actions/jetrace_actions.h"
+#include "actions/jukebox_actions.h"
+#include "dunebuggy.h"
+#include "isle.h"
+#include "legoanimationmanager.h"
+#include "legocontrolmanager.h"
+#include "legohideanimpresenter.h"
+#include "legomain.h"
+#include "legopathstruct.h"
+#include "legoracers.h"
+#include "legoracespecial.h"
+#include "legoutils.h"
+#include "misc.h"
+#include "mxactionnotificationparam.h"
+#include "mxbackgroundaudiomanager.h"
+#include "mxmisc.h"
+#include "mxstillpresenter.h"
+#include "mxtransitionmanager.h"
+#include "mxvariabletable.h"
+#include "scripts.h"
+
+// Defined in legopathstruct.cpp
+extern MxBool g_unk0x100f119c;
+
+// Defined in jetski.cpp
+extern const char* g_varJSFRNTY5;
+extern const char* g_varJSWNSHY5;
+
+// Defined in legoracespecial.cpp
+extern const char* g_raceState;
+extern const char* g_racing;
+
+// Defined in legopathactor.cpp
+extern const char* g_strHIT_WALL_SOUND;
+
 DECOMP_SIZE_ASSERT(JetskiRace, 0x144)
 
-// STUB: LEGO1 0x100162c0
+// FUNCTION: LEGO1 0x100162c0
+// FUNCTION: BETA10 0x100c7e6f
 MxResult JetskiRace::Create(MxDSAction& p_dsAction)
 {
-	return SUCCESS;
+	MxResult result = LegoRace::Create(p_dsAction);
+
+	GameState()->m_currentArea = LegoGameState::e_jetrace;
+	GameState()->StopArea(LegoGameState::e_undefined);
+	LegoGameState* gameState = GameState();
+	RaceState* jetskiRaceState = (RaceState*) gameState->GetState("JetskiRaceState");
+
+	if (!jetskiRaceState) {
+		jetskiRaceState = (RaceState*) gameState->CreateState("JetskiRaceState");
+	}
+
+	m_raceState = jetskiRaceState;
+
+	if (!jetskiRaceState) {
+		return FAILURE;
+	}
+
+	m_raceState->m_unk0x28 = 1;
+	m_unk0x130.SetLeft(397);
+	m_unk0x130.SetTop(317);
+	m_unk0x130.SetRight(543);
+	m_unk0x130.SetBottom(333);
+	LegoRaceCar::FUN_10013670();
+	InvokeAction(
+		Extra::e_start,
+		m_atomId,
+		DuneBuggy::GetColorOffset(g_varJSFRNTY5) + (DuneBuggy::GetColorOffset(g_varJSWNSHY5) * 5 + 0xf) * 2,
+		NULL
+	);
+	InvokeAction(Extra::e_start, m_atomId, 0x61, NULL);
```

Comment:
`0x61` should probably be `JetskiScript::c_JSFRNT_T1_Up_Bitmap`

Reply:
Maybe also `JetraceScript::c_JetskiDashboard`? I left in the `0x61` because no candidate looked really obvious.

Reply:
Ah yes, then I'm pretty sure it's `JetraceScript::c_JetskiDashboard`

Reply:
Ah yes, I'm pretty sure it would have to be that one. It's the same as in `Jetski::HandleClick`